server/src/main/java/keywhiz/service/daos/AclDAO.java

```diff
@@ -135,103 +153,156 @@ public void findAndEvictClient(long clientId, long groupId) {
-            "JOIN memberships ON clients.id = memberships.clientId " +
-            "JOIN accessGrants ON memberships.groupId = accessGrants.groupId " +
-            "JOIN secrets ON secrets.id = accessGrants.secretId " +
-            "WHERE secrets.name = :name")
-  public abstract Set<Client> getClientsFor(@BindSecret Secret secret);
+  public Set<Client> getClientsFor(Secret secret) {
+    List<Client> r = dslContext
+        .select(CLIENTS.ID, CLIENTS.NAME, CLIENTS.DESCRIPTION, CLIENTS.CREATEDAT, CLIENTS.CREATEDBY,
+            CLIENTS.UPDATEDAT, CLIENTS.UPDATEDBY, CLIENTS.ENABLED, CLIENTS.AUTOMATIONALLOWED)
+        .from(CLIENTS)
+        .join(MEMBERSHIPS).on(CLIENTS.ID.eq(MEMBERSHIPS.CLIENTID))
+        .join(ACCESSGRANTS).on(MEMBERSHIPS.GROUPID.eq(ACCESSGRANTS.GROUPID))
+        .join(SECRETS).on(SECRETS.ID.eq(ACCESSGRANTS.SECRETID))
+        .where(SECRETS.NAME.eq(secret.getName()))
+        .fetch()
+        .map(new ClientJooqMapper());
+    return new HashSet<>(r);
+  }
 
-  @Transaction
   public Optional<SanitizedSecret> getSanitizedSecretFor(Client client, String name, String version) {
     checkNotNull(client);
     checkArgument(!name.isEmpty());
     checkNotNull(version);
 
-    Optional<SecretSeries> secretSeries = getSecretSeriesFor(client, name);
-    if (!secretSeries.isPresent()) {
-      return Optional.empty();
-    }
+    // Cast to fix issue with mvn + java8 (not showing up in Intellij).
+    return (Optional<SanitizedSecret>)dslContext.transactionResult(configuration -> {
+      Optional<SecretSeries> secretSeries = getSecretSeriesFor(client, name);
+      if (!secretSeries.isPresent()) {
+        return Optional.empty();
+      }
 
-    Optional<SecretContent> secretContent =
-        createSecretContentDAO().getSecretContentBySecretIdAndVersion(
-            secretSeries.get().getId(), version);
-    if (!secretContent.isPresent()) {
-      return Optional.empty();
-    }
+      Optional<SecretContent> secretContent =
+          secretContentDAO.getSecretContentBySecretIdAndVersion(
+              secretSeries.get().getId(), version);
+      if (!secretContent.isPresent()) {
+        return Optional.empty();
+      }
 
-    SecretSeriesAndContent seriesAndContent = SecretSeriesAndContent.of(
-        secretSeries.get(), secretContent.get());
-    return Optional.of(SanitizedSecret.fromSecretSeriesAndContent(seriesAndContent));
+      SecretSeriesAndContent seriesAndContent = SecretSeriesAndContent.of(
+          secretSeries.get(), secretContent.get());
+      return Optional.of(SanitizedSecret.fromSecretSeriesAndContent(seriesAndContent));
+    });
   }
 
-  @SqlUpdate("INSERT INTO accessGrants (secretId, groupId) VALUES (:secretId, :groupId)")
-  protected abstract void allowAccess(@Bind("secretId") long secretId, @Bind("groupId") long groupId);
+  protected void allowAccess(long secretId, long groupId) {
+    dslContext
+        .insertInto(ACCESSGRANTS)
+        .set(ACCESSGRANTS.SECRETID, (int) secretId)
```

Comment:
Sure. We do go from int -> long -> int, so it's not possible for things to overflow.